jina/helper.py

```diff
@@ -1322,9 +1333,10 @@ def run(self):
                 'please report this issue here: https://github.com/jina-ai/jina'
             )
     else:
-        loop = get_or_reuse_loop()
+        loop, is_new_loop = get_or_reuse_loop_with_info()
         result = loop.run_until_complete(func(*args, **kwargs))
```

Comment:
wrap in context manager or something to guarantee is closed if something wrong happens only if is owned